File: frontend/src/lib/types/metrics.ts
export interface MetricsResponse {
  total_leads: number;
  total_acionamentos: number;
  por_status: Record<string, number>;
  por_canal: Record<string, number>;
  taxa_conversao: number;
  taxa_resposta: number;
}

export interface AgentMetricsRow {
  agent_id: string;
  agent_name: string;
  mode: string;
  total_leads: number;
  total_acionamentos: number;
  por_status: Record<string, number>;
  por_canal: Record<string, number>;
  taxa_conversao: number;
  taxa_resposta: number;
}

export interface AgentMetricsResponse {
  agents: AgentMetricsRow[];
}

export const AGENT_MODE_LABELS: Record<string, string> = {
  ACTIVE: "Ativo",
  RECEPTIVE: "Receptivo",
};

export interface DashboardCards {
  agents: number;
  active_channels: number;
  leads: number;
  active_campaigns: number;
}

export interface DashboardSummaryResponse {
  cards: DashboardCards;
  leads_acionados: number;
  leads_virgens: number;
  tentativas_por_canal: Record<string, number>;
  tentativas_por_status: Record<string, number>;
}

/** Filtro de canal — seletor na home do dashboard. */
export type DashboardChannelFilter = "whatsapp" | "telegram" | "voice" | null;

export interface DashboardCampaignRow {
  campaign_id: string;
  campaign_name: string;
  leads: number;
  data_recebimento: string | null;
  data_inicio: string | null;
  data_fim: string | null;
  tentativas: number;
  spin: number;
  contato: number;
  cpc: number;
  sucesso: number;
  conversao: number;
}

export interface DashboardCampaignsResponse {
  campaigns: DashboardCampaignRow[];
}

export const STATUS_ORDER = [
  "pendente",
  "acionado",
  "em_andamento",
  "nao_atendido",
  "convertido",
  "recusou",
  "erro",
] as const;

export const STATUS_LABELS: Record<string, string> = {
  pendente: "Pendente",
  acionado: "Acionado",
  em_andamento: "Em andamento",
  nao_atendido: "Não atendido",
  convertido: "Convertido",
  recusou: "Recusou",
  erro: "Erro",
};

export const STATUS_COLORS: Record<string, string> = {
  convertido: "#22c55e",
  recusou: "#ef4444",
  em_andamento: "#3b82f6",
  nao_atendido: "#6b7280",
  acionado: "#eab308",
  erro: "#f97316",
  pendente: "#d1d5db",
};

export const CHANNEL_LABELS: Record<string, string> = {
  whatsapp: "WhatsApp",
  telegram: "Telegram",
  voice: "Voz",
};

export const CHANNEL_COLORS: Record<string, string> = {
  whatsapp: "#25D366",
  telegram: "#0088cc",
  voice: "#8b5cf6",
};

export interface ChannelQueueMetrics {
  total_enfileirados: number;
  total_atendidos: number;
  total_abandonados: number;
  tempo_medio_espera: number;
  taxa_abandono: number;
  nivel_servico: number;
  tamanho_fila_atual: number;
}

export interface QueueMetricsResponse {
  period_days: number;
  service_level_target_seconds: number;
  total_enfileirados: number;
  total_atendidos: number;
  total_abandonados: number;
  tempo_medio_espera: number;
  taxa_abandono: number;
  nivel_servico: number;
  tamanho_fila_atual: number;
  por_canal: Record<string, ChannelQueueMetrics>;
  abandono_apenas_voz: boolean;
  abandono_disponivel_inbound: boolean;
}

File: backend/app/schemas/metrics.py
"""Pydantic schemas for campaign and lead base metrics."""

from uuid import UUID

from pydantic import BaseModel, Field


class MetricsResponse(BaseModel):
    total_leads: int
    total_acionamentos: int
    por_status: dict[str, int] = Field(
        default_factory=lambda: {
            "pendente": 0,
            "acionado": 0,
            "em_andamento": 0,
            "nao_atendido": 0,
            "convertido": 0,
            "recusou": 0,
            "erro": 0,
        }
    )
    por_canal: dict[str, int] = Field(
        default_factory=lambda: {
            "whatsapp": 0,
            "telegram": 0,
            "voice": 0,
        }
    )
    taxa_conversao: float
    taxa_resposta: float


class AgentMetricsRow(BaseModel):
    agent_id: UUID
    agent_name: str
    mode: str
    total_leads: int
    total_acionamentos: int
    por_status: dict[str, int] = Field(
        default_factory=lambda: {
            "pendente": 0,
            "acionado": 0,
            "em_andamento": 0,
            "nao_atendido": 0,
            "convertido": 0,
            "recusou": 0,
            "erro": 0,
        }
    )
    por_canal: dict[str, int] = Field(
        default_factory=lambda: {
            "whatsapp": 0,
            "telegram": 0,
            "voice": 0,
        }
    )
    taxa_conversao: float
    taxa_resposta: float


class AgentMetricsResponse(BaseModel):
    agents: list[AgentMetricsRow]


class ChannelQueueMetrics(BaseModel):
    """Métricas de fila receptiva por canal (R-B)."""

    total_enfileirados: int = 0
    total_atendidos: int = 0
    total_abandonados: int = 0
    tempo_medio_espera: float = 0.0
    taxa_abandono: float = 0.0
    nivel_servico: float = 0.0
    tamanho_fila_atual: int = 0


class QueueMetricsResponse(BaseModel):
    """
    Métricas agregadas da fila de atendimento receptivo.

    Abandono (taxa_abandono) aplica-se só a VOZ; sem inbound de voz os valores
    tendem a zero — ver abandono_disponivel_inbound.
    """

    period_days: int = 1
    service_level_target_seconds: int = 20
    total_enfileirados: int = 0
    total_atendidos: int = 0
    total_abandonados: int = 0
    tempo_medio_espera: float = 0.0
    taxa_abandono: float = 0.0
    nivel_servico: float = 0.0
    tamanho_fila_atual: int = 0
    por_canal: dict[str, ChannelQueueMetrics] = Field(default_factory=dict)
    abandono_apenas_voz: bool = True
    abandono_disponivel_inbound: bool = False
